src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,197 @@
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SOUL_MAX_VALUE = 100
+const SHROUD_MAX_VALUE = 100
+
+const SOUL_GAUGE_COLOR = Color('#DC0707FF')
+const SHROUD_GAUGE_COLOR = Color('#2C9FCB')
+
+const SOUL_GENERATORS: ActionKey[] = [
+	'SLICE',
+	'WAXING_SLICE',
+	'INFERNAL_SLICE',
+	'SPINNING_SCYTHE',
+	'NIGHTMARE_SCYTHE',
+	'SOUL_SLICE',
+	'SOUL_SCYTHE',
+]
+
+const SOUL_CONSUMERS: ActionKey[] = [
+	'BLOOD_STALK',
+	'GRIM_SWATHE',
+	'GLUTTONY',
+	'UNVEILED_GIBBET',
+	'UNVEILED_GALLOWS',
+]
+
+const SHROUD_GENERATORS: ActionKey[] = [
+	'GIBBET',
+	'GALLOWS',
+	'GUILLOTINE',
+	'PLENTIFUL_HARVEST',
+]
+
+const SHROUD_CONSUMERS: ActionKey[] = [
+	'ENSHROUD',
+]
+
+/**
+ * Base amount for the combo actions
+ */
+const BASE_GAUGE_GENERATION_AMOUNT = 10
+
+/**
+ * Base amount for the soul gauge consumption
+ */
+const BASE_GAUGE_CONSUMPTION_AMOUNT = 50
+const PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT = 50
+const ENSHROUD_GAUGE_CONSUMPTION = 50
+
+/**
+ * Amount generated for the "Soul" skills, which are cooldowns
+ */
+const SOUL_GAUGE_GENERATION_AMOUNT = 50
+
+export class Gauge extends CoreGauge {
+	@dependency private suggestions!: Suggestions
+
+	private soulGeneratorModifiers = new Map<number, number>([
+		[this.data.actions.SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.WAXING_SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.INFERNAL_SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SPINNING_SCYTHE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.NIGHTMARE_SCYTHE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SOUL_SLICE.id, SOUL_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SOUL_SCYTHE.id, SOUL_GAUGE_GENERATION_AMOUNT],
+	])
+
+	private soulConsumptionModifiers = new Map<number, number>([
+		[this.data.actions.BLOOD_STALK.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.GRIM_SWATHE.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.GLUTTONY.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.UNVEILED_GIBBET.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.UNVEILED_GALLOWS.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+	])
+
+	private shroudGeneratorModifiers = new Map<number, number>([
+		[this.data.actions.GIBBET.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.GALLOWS.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.GUILLOTINE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.PLENTIFUL_HARVEST.id, PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT],
+	])
+
+	private shroudConsumerModifiers = new Map<number, number>([
+		[this.data.actions.ENSHROUD.id, ENSHROUD_GAUGE_CONSUMPTION],
+	])
```

Comment:
I would personally replace all of this by adding a couple extra event hooks and altering your actionKey[]s, since there's a lot of repetition here, when what you have is mostly default behavior, with a few edge cases.